experiments/vectorized/concept.py

```diff
@@ -213,15 +227,20 @@ def compute_self_att_loss(alphas):
     def train(epoch=-1, backprop=True):
         """ Train model for a single epoch.
         """
+        ### Model is declared in global space 
```

Comment:
Yup, this is a [function closure](https://en.wikipedia.org/wiki/Closure_(computer_programming))